File: Lab 1/src/main/java/lexer/Token.java
package lexer;

public class Token {
    public enum Type {
        COMMENT,
        WHITESPACE,
        IDENTIFIER,
        OPERATOR,
        SEPARATOR,
        INT_LITERAL,
        FLOAT_LITERAL,
        CHAR_LITERAL,
        STRING_LITERAL,
        BOOLEAN_LITERAL,
        NULL_LITERAL,
        KEYWORD,
        ERROR
    }

    private final Type type;

    private final String tokenString;

    Token(String tokenString, Type type) {
        this.type = type;
        this.tokenString = tokenString;
    }

    public String getTokenString() {
        return this.tokenString;
    }

    public Type getType() {
        return this.type;
    }

    @Override
    public String toString() {
        return "Token{" +
                "type=" + type +
                ", tokenString='" + tokenString + '\'' +
                '}';
    }
}

File: Lab 1/src/main/java/lexer/Token.kt
package lexer

class Token internal constructor(val tokenString: String, val type: Type) {
    enum class Type {
        COMMENT, WHITESPACE, IDENTIFIER, OPERATOR, SEPARATOR, INT_LITERAL, FLOAT_LITERAL, CHAR_LITERAL, STRING_LITERAL, BOOLEAN_LITERAL, NULL_LITERAL, KEYWORD, ERROR
    }

    override fun toString(): String {
        return "Token{" +
                "type=" + type +
                ", tokenString='" + tokenString + '\'' +
                '}'
    }

}